pkg/network/event_common.go

```diff
@@ -416,3 +420,48 @@ func generateConnectionKey(c ConnectionStats, buf []byte, useNAT bool) []byte {
 	n += raddr.WriteTo(buf[n:]) // 4 or 16 bytes
 	return buf[:n]
 }
+
+// Sub returns s-other
+func (s StatCounters) Sub(other StatCounters) (sc StatCounters, underflow bool) {
+	if s.RecvBytes < other.RecvBytes ||
+		s.RecvPackets < other.RecvPackets ||
+		(s.Retransmits < other.Retransmits && s.Retransmits > 0) ||
+		s.SentBytes < other.SentBytes ||
+		s.SentPackets < other.SentPackets ||
+		(s.TCPClosed < other.TCPClosed && s.TCPClosed > 0) ||
+		(s.TCPEstablished < other.TCPEstablished && s.TCPEstablished > 0) {
+		return sc, true
+	}
+
+	sc = StatCounters{
+		RecvBytes:   s.RecvBytes - other.RecvBytes,
+		RecvPackets: s.RecvPackets - other.RecvPackets,
+		SentBytes:   s.SentBytes - other.SentBytes,
+		SentPackets: s.SentPackets - other.SentPackets,
+	}
+
+	if s.Retransmits > 0 {
```

Comment:
Why the special treatment for retransmits, etc.? They should be monotonic counters as well, right?

Reply:
Because of this code: https://github.com/DataDog/datadog-agent/pull/12648/files/a7ca2a0cf1bb26f722af3d7c19fb3ca98f7407d1#diff-b3a5827b0c9e4fad1178e080974122e0aca65a7e724645f7c6e8dd80407ee17aR455-R456

Since we can't tell whether retransmits are actually `0` or were reset by us to `0` because of a pid collision.